File: backend/src/main/java/tfg/proyecto/TFG/dtos/DTOCarritoBajada.java
package tfg.proyecto.TFG.dtos;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import tfg.proyecto.TFG.modelo.EstadoCarrito;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder

public class DTOCarritoBajada {
	 private Long id;
	    private EstadoCarrito estado;
	    private List<DTOCarritoItemBajada> items;
}

File: frontend/appMovilShowPass/app/src/main/java/com/example/appmovilshowpass/data/remote/dto/DTOCarritoBajada.kt
package com.example.appmovilshowpass.data.remote.dto

import com.example.appmovilshowpass.model.Carrito
import com.example.appmovilshowpass.model.EstadoCarrito

data class DTOCarritoBajada(
    val id: Long,
    val estado: EstadoCarrito,
    val items: List<DTOCarritoItemBajada>
)


fun DTOCarritoBajada.toCarrito(): Carrito {
    return Carrito(
        id = id,
        estado = estado,
        items = items.map { it.toCarritoItem() }
    )
}